File: rails/app/models/card_ruling.rb
class CardRuling < ApplicationRecord
  self.table_name = 'card_rulings'

  belongs_to :card, class_name: 'Card'

  validates :source, presence: true, length: { maximum: 200 }

  def to_s
    ruling_text.to_s
  end

  # Business operations

  def is_current
    # TODO: implement is_current
    nil
  end

  def supersedes_previous
    # TODO: implement supersedes_previous
    nil
  end
end

File: laravel/app/Models/Cards/CardRuling.php
<?php

namespace App\Models\Cards;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;

class CardRuling extends Model
{
    protected $table = 'card_rulings';

    protected $fillable = ['ruling_text', 'published_at', 'source', 'card_id'];

    protected $casts = [
        'published_at' => 'date',
    ];

    public function card(): BelongsTo
    {
        return $this->belongsTo(Card::class, 'card_id');
    }

    // ── Business operations ──────────────────────────────────────────

    public function isCurrent(): bool
    {
        throw new \RuntimeException('is_current not implemented');
    }

    public function supersedesPrevious(): bool
    {
        throw new \RuntimeException('supersedes_previous not implemented');
    }

}
